test/include/test_runner/test_case.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "main/kuzu.h"
+
+using namespace kuzu::main;
+
+namespace kuzu {
+namespace testing {
+
+struct TestStatement {
+    std::string name;
+    std::string query;
+    std::string blockName;
+    uint64_t numThreads = 4;
+    std::string encodedJoin;
+    uint64_t expectedNumTuples = 0;
+    bool expectedError = false;
+    bool expectedOk = false;
+    std::vector<std::string> expectedTuples;
+    std::string errorMessage;
+    bool enumerate = false;
+    bool checkOutputOrder = false;
+};
+
+struct TestCase {
+    std::string group;
+    std::string name;
+    std::string dataset;
+    std::vector<std::unique_ptr<TestStatement>> statements;
+    std::vector<std::unique_ptr<TestStatement>> variableStatements;
```

Comment:
My suggestion for `variableStatements` is use `unordered_map<string, std::vector<std::unique_ptr<TestStatement>>>`, so when you need to extract a statement block, you can access based on the variable name, instead of looping all statements.